client/components/forms/counted-textarea/index.jsx

```diff
@@ -33,39 +33,36 @@ module.exports = React.createClass( {
 		};
 	},
 
-	renderCountPanel: function() {
-		var length = this.props.value.length;
-
+	renderCountPanel() {
+		let length = this.props.value.length;
 		if ( ! length && this.props.countPlaceholderLength ) {
 			length = this.props.placeholder.length;
 		}
 
+		let panelText = this.translate( '%d character', '%d characters', {
```

Comment:
It's a minor optimization, but feels wasteful to call `translate` only to be overridden in some conditions. Consider assigning in an `else` block:

```jsx
let panelText;
if ( this.props.showRemainingCharacters && this.props.acceptableLength ) {
	panelText = this.translate( '%d character remaining', '%d characters remaining', {
		context: 'Input length',
		args: [ this.props.acceptableLength - length ],
		count: this.props.acceptableLength - length
	} );
} else {
	panelText = this.translate( '%d character', '%d characters', {
		context: 'Input length',
		args: [ length ],
		count: length
	} );
}
```